tuna.go

```diff
@@ -311,6 +318,24 @@ func (c *Common) CreateServerConn(force bool) error {
 
 				metadata := c.GetMetadata()
 
+				if !c.EntryServiceInfo.IPFilter.Empty() {
+					if !c.EntryServiceInfo.IPFilter.IPCheck(metadata.IP) {
+						continue
+					}
+					if !c.EntryServiceInfo.IPFilter.Location.Empty() {
+						loc, err := getLocationFromIP2C(metadata.IP)
+						if err != nil {
+							log.Println(err)
+							c.EntryServiceInfo.IPFilter.Location = nil
```

Comment:
This will lead to unwanted results: if ip2c service is down for a few minutes, then recovered, it will never do ip filter again